Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -108,36 +97,76 @@ - (NSURLSessionTaskState)state
 
 #pragma mark Helpers
 
+- (NSURLSessionDataTask *)makeRequest
+{
+    __block CDTHTTPInterceptorContext *ctx =
+        [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    // We make sure all objects support `interceptRequestInContext:` during init.
+    for (NSObject<CDTHTTPInterceptor> *obj in self.requestInterceptors) {
+        ctx = [obj interceptRequestInContext:ctx];
+    }
+
+    __weak CDTURLSessionTask *weakSelf = self;
+    return [self.session
+        dataTaskWithRequest:ctx.request
+          completionHandler:^void(NSData *data, NSURLResponse *response, NSError *error) {
+            __strong CDTURLSessionTask *strongSelf = weakSelf;
+            if (strongSelf) {
+                ctx.response = response;
+                for (NSObject<CDTHTTPInterceptor> *obj in strongSelf.responseInterceptors) {
+                    ctx = [obj interceptResponseInContext:ctx];
+                }
+
+                if (ctx.shouldRetry && strongSelf.remainingRetries > 0) {
+                    // retry
+                    strongSelf.remainingRetries--;
+                    strongSelf.inProgressTask = [strongSelf makeRequest];
+                    [strongSelf.inProgressTask resume];
+                } else if (strongSelf.completionHandler) {
+                    data = [NSData dataWithData:data];
+                    strongSelf.completionHandler(data, response, error);
+                }
+            }
+          }];
+}
+
 /**
  Copy the interceptor array, filtering out non-compliant classes.
 
- We do this once during -init... to avoid spamming the logs for every retry. This checks for
- responding to `interceptRequestInContext:` as we're creating a request interceptor array,
- not a response one.
+ We do this once during `-init...`. This checks for responding to `interceptRequestInContext:`
+ as we're creating a request interceptor array, not a response one.
  */
 - (NSArray *)filterRequestInterceptors:(NSArray *)proposedRequestInterceptors
 {
-    NSMutableArray *acc = [NSMutableArray array];
+    NSMutableArray *requestInterceptors = [NSMutableArray array];
 
     for (NSObject *obj in proposedRequestInterceptors) {
-        if (![obj conformsToProtocol:@protocol(CDTHTTPInterceptor)]) {
-            CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't conform to protocol \
-                       CDTURLSessionInterceptor, skipping",
-                       obj);
-            continue;
+        if ([obj respondsToSelector:@selector(interceptRequestInContext:)]) {
```

Comment:
This change drops the warning message that was previously in place. I think it would be useful to leave this logging in the code as it might help a confused user who hasn't set their interceptors up correctly.